src/Microsoft.ML.TensorFlow/TensorFlow/Tensor.cs

```diff
@@ -410,9 +410,9 @@ public unsafe TFTensor(long value)
         public TFTensor(Complex[] data) : base(SetupTensor(TFDataType.Complex128, data, size: 16)) { }
 
         // Convenience function to factor out the setup of a new tensor from an array
-        internal static IntPtr SetupTensor(TFDataType dt, long[] dims, Array data, int size)
+        internal static IntPtr SetupTensor(TFDataType dt, long[] dims, Array data, int count, int size)
         {
-            return SetupTensor(dt, dims, data, start: 0, count: data.Length, size: size);
+            return SetupTensor(dt, dims, data, start: 0, count: count, size: size);
```

Comment:
>start: 0, count: count, size: size [](start = 47, length = 34)

nit: do you need to specify param names here? don't you invoke function with all params already